backend/battling/views.py

```diff
@@ -71,26 +90,62 @@ def round_new2(request):
     if request.method == "POST":
         form_round2 = RoundForm2(request.POST, instance=battle_info)
         if form_round2.is_valid():
-            teste = form_round2.save(commit=False)
-            data_pk21 = get_pokemon_from_api(teste.pk21)
-            data_pk22 = get_pokemon_from_api(teste.pk22)
-            data_pk23 = get_pokemon_from_api(teste.pk23)
-            sum_pk21 = sumValid(data_pk21)
-            sum_pk22 = sumValid(data_pk22)
-            sum_pk23 = sumValid(data_pk23)
-            sum_all = sum_pk21 + sum_pk22 + sum_pk23
-            if sum_all <= 600:
+            round_battle = form_round2.save(commit=False)
+            valid_team = check_valid_team2(round_battle)
+            if valid_team:
                 form_round2.save()
-                return redirect("home")
-            if sum_all > 600:
-                message = "ERROR: The PKNs sum more than 600 pts. Please choose again"
-                return render(
-                    request,
-                    "battling/round_new2.html",
-                    {"form_round2": form_round2, "battle": battle_info, "message": message},
-                )
+                return redirect("battle_end")
+            message = "ERROR: your PKNs sum more than 600 points"
+            return render(
+                request,
+                "battling/round_new2.html",
+                {"form_round2": form_round2, "battle": battle_info, "message": message},
+            )
     else:
         form_round2 = RoundForm2()
     return render(
         request, "battling/round_new2.html", {"form_round2": form_round2, "battle": battle_info}
     )
+
+
+def battle_end(request):
+    battle_id = Battle.objects.latest("id").id
+    battle_info = Battle.objects.filter(id=battle_id).values()[0]
+    id_battle = Battle.objects.get(id=battle_id)
+
+    creator_pkms = [get_pokemon_from_api(battle_info["pk1" + str(i)]) for i in range(1, 4)]
+    opponent_pkms = [get_pokemon_from_api(battle_info["pk2" + str(i)]) for i in range(1, 4)]
+
+    score = battle(creator_pkms, opponent_pkms)
+
+    winner = (
+        id_battle.player1.email if score["creator"] > score["opponent"] else id_battle.player2.email
+    )
+    send_battle_result(id_battle, winner, creator_pkms, opponent_pkms)
+
+    return render(request, "battling/battle_end.html")
+
+
+def fights(request):
+    battle_id = Battle.objects.latest("id").id
+    battle_info = Battle.objects.filter(id=battle_id).values()[0]
+    id_battle = Battle.objects.get(id=battle_id)
```

Comment:
Same as above

Reply:
Niceee... Will do the changes! Thank you, Renato!!!